extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/QuarkusPathLocationScanner.java

```diff
@@ -0,0 +1,141 @@
+package io.quarkus.flyway.runtime;
+
+import java.io.IOException;
+import java.net.JarURLConnection;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import org.flywaydb.core.api.Location;
+import org.flywaydb.core.api.logging.Log;
+import org.flywaydb.core.api.logging.LogFactory;
+import org.flywaydb.core.internal.resource.LoadableResource;
+import org.flywaydb.core.internal.resource.classpath.ClassPathResource;
+import org.flywaydb.core.internal.scanner.classpath.ResourceAndClassScanner;
+
+public final class QuarkusPathLocationScanner implements ResourceAndClassScanner {
+    private static final Log LOG = LogFactory.getLog(QuarkusPathLocationScanner.class);
+    // TODO: this should be configurable, using flyway default
+    private final static String DEFAULT_NATIVE_LOCATION = "db/migration";
+    private static Set<String> ALL_SQL_RESOURCES;
+
+    static {
+        try {
+            ALL_SQL_RESOURCES = discoverApplicationMigrations();
+        } catch (IOException | URISyntaxException e) {
+            throw new IllegalStateException("Could not discover Flyway migrations", e);
+        }
+    }
+
+    public QuarkusPathLocationScanner(Location location) {
+        if (!location.getPath().equals(DEFAULT_NATIVE_LOCATION)) {
+            LOG.error("Invalid migrations location: " + location.getPath() + ". Flyway migrations must be located in  "
+                    + DEFAULT_NATIVE_LOCATION);
+        }
+    }
+
+    private static Set<String> discoverApplicationMigrations() throws IOException, URISyntaxException {
+        Set<String> resources = new HashSet<>();
+        try {
+            Enumeration<URL> migrations = Thread.currentThread().getContextClassLoader().getResources(
+                    DEFAULT_NATIVE_LOCATION);
+            while (migrations.hasMoreElements()) {
+                URL path = migrations.nextElement();
+                LOG.info("Adding application migrations in path: " + path);
```

Comment:
This should be debug, we are tying to keep the default logging minimal, so any potential issues are easy to spot.